File: backend/src/main/java/com/nammaooru/service/EmailService.java
package com.nammaooru.service;

import lombok.RequiredArgsConstructor;
import lombok.extern.slf4j.Slf4j;
import org.springframework.beans.factory.annotation.Value;
import org.springframework.mail.javamail.JavaMailSender;
import org.springframework.mail.javamail.MimeMessageHelper;
import org.springframework.scheduling.annotation.Async;
import org.springframework.stereotype.Service;
import org.thymeleaf.TemplateEngine;
import org.thymeleaf.context.Context;

import javax.mail.MessagingException;
import javax.mail.internet.MimeMessage;
import java.math.BigDecimal;
import java.time.LocalDate;
import java.time.LocalDateTime;
import java.time.format.DateTimeFormatter;
import java.util.HashMap;
import java.util.List;
import java.util.Map;

@Slf4j
@Service
@RequiredArgsConstructor
public class EmailService {

    private final JavaMailSender mailSender;
    private final TemplateEngine templateEngine;
    
    @Value("${spring.mail.username}")
    private String fromEmail;
    
    @Value("${app.name:NammaOoru}")
    private String appName;

    // Send OTP email
    @Async
    public void sendOTPEmail(String to, String otp, String purpose) {
        try {
            String subject = getOTPSubject(purpose);
            String body = generateOTPEmailBody(otp, purpose);
            sendHtmlEmail(to, subject, body);
            log.info("OTP email sent to {} for {}", to, purpose);
        } catch (Exception e) {
            log.error("Failed to send OTP email to {}: ", to, e);
        }
    }

    // Send order confirmation email
    @Async
    public void sendOrderConfirmation(String to, String orderNumber, String customerName, String deliveryOTP) {
        try {
            String subject = String.format("Order Confirmed - #%s", orderNumber);
            Map<String, Object> variables = new HashMap<>();
            variables.put("orderNumber", orderNumber);
            variables.put("customerName", customerName);
            variables.put("deliveryOTP", deliveryOTP);
            variables.put("appName", appName);
            
            String body = generateOrderConfirmationEmail(variables);
            sendHtmlEmail(to, subject, body);
            log.info("Order confirmation sent to {} for order {}", to, orderNumber);
        } catch (Exception e) {
            log.error("Failed to send order confirmation to {}: ", to, e);
        }
    }

    // Send invoice email
    @Async
    public void sendInvoiceEmail(String to, InvoiceData invoice) {
        try {
            String subject = String.format("Invoice - Order #%s", invoice.getOrderNumber());
            String body = generateInvoiceHTML(invoice);
            sendHtmlEmail(to, subject, body);
            log.info("Invoice email sent to {} for order {}", to, invoice.getOrderNumber());
        } catch (Exception e) {
            log.error("Failed to send invoice to {}: ", to, e);
        }
    }

    // Send order status update
    @Async
    public void sendOrderStatusUpdate(String to, String orderNumber, String status, String message, String otp) {
        try {
            String subject = String.format("Order %s - %s", orderNumber, getStatusTitle(status));
            Map<String, Object> variables = new HashMap<>();
            variables.put("orderNumber", orderNumber);
            variables.put("status", status);
            variables.put("message", message);
            variables.put("otp", otp);
            
            String body = generateStatusUpdateEmail(variables);
            sendHtmlEmail(to, subject, body);
            log.info("Status update sent to {} for order {} - {}", to, orderNumber, status);
        } catch (Exception e) {
            log.error("Failed to send status update to {}: ", to, e);
        }
    }

    // Send daily summary to shop owner
    @Async
    public void sendDailySummary(String to, DailySummaryData summaryData) {
        try {
            String subject = String.format("Daily Summary - %s - %s", 
                summaryData.getShopName(), summaryData.getDate());
            String body = generateDailySummaryHTML(summaryData);
            sendHtmlEmail(to, subject, body);
            log.info("Daily summary sent to {} for shop {}", to, summaryData.getShopName());
        } catch (Exception e) {
            log.error("Failed to send daily summary to {}: ", to, e);
        }
    }

    // Send delivery notification
    @Async
    public void sendDeliveryNotification(String to, String orderNumber, String trackingUrl, String deliveryOTP) {
        try {
            String subject = String.format("Your order %s is out for delivery!", orderNumber);
            Map<String, Object> variables = new HashMap<>();
            variables.put("orderNumber", orderNumber);
            variables.put("trackingUrl", trackingUrl);
            variables.put("deliveryOTP", deliveryOTP);
            
            String body = generateDeliveryNotificationEmail(variables);
            sendHtmlEmail(to, subject, body);
            log.info("Delivery notification sent to {} for order {}", to, orderNumber);
        } catch (Exception e) {
            log.error("Failed to send delivery notification to {}: ", to, e);
        }
    }

    // Core email sending method
    private void sendHtmlEmail(String to, String subject, String htmlBody) throws MessagingException {
        MimeMessage message = mailSender.createMimeMessage();
        MimeMessageHelper helper = new MimeMessageHelper(message, true, "UTF-8");
        
        helper.setFrom(fromEmail);
        helper.setTo(to);
        helper.setSubject(subject);
        helper.setText(htmlBody, true);
        
        mailSender.send(message);
    }

    // Generate OTP email body
    private String generateOTPEmailBody(String otp, String purpose) {
        return String.format("""
            <!DOCTYPE html>
            <html>
            <head>
                <style>
                    body { font-family: Arial, sans-serif; background: #f5f5f5; padding: 20px; }
                    .container { max-width: 600px; margin: 0 auto; background: white; border-radius: 10px; padding: 30px; }
                    .otp-box { background: #f0f9ff; border: 2px solid #0ea5e9; border-radius: 8px; padding: 20px; text-align: center; margin: 20px 0; }
                    .otp-code { font-size: 32px; font-weight: bold; color: #0ea5e9; letter-spacing: 5px; }
                    .purpose { color: #64748b; margin-bottom: 20px; }
                </style>
            </head>
            <body>
                <div class="container">
                    <h2>%s - OTP Verification</h2>
                    <p class="purpose">%s</p>
                    <div class="otp-box">
                        <p>Your OTP code is:</p>
                        <div class="otp-code">%s</div>
                        <p style="color: #ef4444; margin-top: 15px;">This OTP is valid for 10 minutes</p>
                    </div>
                    <p style="color: #6b7280;">If you didn't request this, please ignore this email.</p>
                </div>
            </body>
            </html>
            """, appName, getPurposeMessage(purpose), otp);
    }

    // Generate order confirmation email
    private String generateOrderConfirmationEmail(Map<String, Object> variables) {
        String orderNumber = (String) variables.get("orderNumber");
        String customerName = (String) variables.get("customerName");
        String deliveryOTP = (String) variables.get("deliveryOTP");
        
        return String.format("""
            <!DOCTYPE html>
            <html>
            <head>
                <style>
                    body { font-family: Arial, sans-serif; background: #f5f5f5; padding: 20px; }
                    .container { max-width: 600px; margin: 0 auto; background: white; border-radius: 10px; overflow: hidden; }
                    .header { background: #10b981; color: white; padding: 30px; text-align: center; }
                    .content { padding: 30px; }
                    .otp-section { background: #f0fdf4; border-left: 4px solid #10b981; padding: 15px; margin: 20px 0; }
                    .otp-code { font-size: 24px; font-weight: bold; color: #10b981; }
                </style>
            </head>
            <body>
                <div class="container">
                    <div class="header">
                        <h1>Order Confirmed! ✅</h1>
                        <p>Order #%s</p>
                    </div>
                    <div class="content">
                        <p>Dear %s,</p>
                        <p>Your order has been confirmed and will be prepared soon.</p>
                        
                        <div class="otp-section">
                            <p><strong>Delivery OTP:</strong></p>
                            <div class="otp-code">%s</div>
                            <p style="color: #6b7280; font-size: 14px;">Share this OTP with delivery partner upon delivery</p>
                        </div>
                        
                        <p>You will receive updates as your order progresses.</p>
                        <p>Thank you for ordering with %s!</p>
                    </div>
                </div>
            </body>
            </html>
            """, orderNumber, customerName, deliveryOTP, appName);
    }

    // Generate invoice HTML
    private String generateInvoiceHTML(InvoiceData invoice) {
        StringBuilder itemsHtml = new StringBuilder();
        for (InvoiceItem item : invoice.getItems()) {
            itemsHtml.append(String.format("""
                <tr>
                    <td>%s</td>
                    <td>%d %s</td>
                    <td>₹%.2f</td>
                    <td>₹%.2f</td>
                </tr>
                """, item.getProductName(), item.getQuantity(), item.getUnit(), 
                item.getPrice(), item.getTotal()));
        }
        
        return String.format("""
            <!DOCTYPE html>
            <html>
            <head>
                <style>
                    body { font-family: Arial, sans-serif; }
                    .invoice { max-width: 800px; margin: 0 auto; padding: 20px; }
                    .header { background: #2563eb; color: white; padding: 20px; text-align: center; }
                    table { width: 100%%; border-collapse: collapse; margin: 20px 0; }
                    th, td { padding: 10px; text-align: left; border-bottom: 1px solid #ddd; }
                    th { background: #f8f9fa; }
                    .total { font-size: 20px; font-weight: bold; color: #10b981; }
                </style>
            </head>
            <body>
                <div class="invoice">
                    <div class="header">
                        <h1>%s - Invoice</h1>
                        <p>Order #%s</p>
                    </div>
                    <p><strong>Date:</strong> %s</p>
                    <p><strong>Customer:</strong> %s</p>
                    <p><strong>Delivery Address:</strong> %s</p>
                    
                    <table>
                        <thead>
                            <tr>
                                <th>Item</th>
                                <th>Quantity</th>
                                <th>Price</th>
                                <th>Total</th>
                            </tr>
                        </thead>
                        <tbody>%s</tbody>
                    </table>
                    
                    <div style="text-align: right;">
                        <p>Subtotal: ₹%.2f</p>
                        <p>Delivery Fee: ₹%.2f</p>
                        %s
                        <p class="total">Total: ₹%.2f</p>
                    </div>
                    
                    <p style="text-align: center; color: #6b7280; margin-top: 40px;">
                        Thank you for your order!
                    </p>
                </div>
            </body>
            </html>
            """, appName, invoice.getOrderNumber(), 
            invoice.getOrderDate().format(DateTimeFormatter.ofPattern("dd/MM/yyyy HH:mm")),
            invoice.getCustomerName(), invoice.getDeliveryAddress(),
            itemsHtml.toString(), invoice.getSubtotal(), invoice.getDeliveryFee(),
            invoice.getDiscount().compareTo(BigDecimal.ZERO) > 0 ? 
                String.format("<p style='color: #10b981;'>Discount: -₹%.2f</p>", invoice.getDiscount()) : "",
            invoice.getTotal());
    }

    // Generate daily summary HTML
    private String generateDailySummaryHTML(DailySummaryData data) {
        // Generate top items HTML
        StringBuilder topItemsHtml = new StringBuilder();
        for (TopSellingItem item : data.getTopSellingItems()) {
            topItemsHtml.append(String.format("""
                <tr>
                    <td>%s</td>
                    <td>%d</td>
                    <td>₹%.2f</td>
                </tr>
                """, item.getName(), item.getQuantity(), item.getRevenue()));
        }
        
        return String.format("""
            <!DOCTYPE html>
            <html>
            <head>
                <style>
                    body { font-family: Arial, sans-serif; background: #f5f5f5; }
                    .container { max-width: 800px; margin: 0 auto; background: white; padding: 20px; }
                    .header { background: linear-gradient(135deg, #667eea 0%%, #764ba2 100%%); color: white; padding: 30px; text-align: center; }
                    .stats-grid { display: grid; grid-template-columns: repeat(2, 1fr); gap: 20px; margin: 20px 0; }
                    .stat-card { background: #f8fafc; padding: 20px; border-radius: 8px; }
                    .profit-section { background: #f0fdf4; padding: 20px; border-radius: 8px; border-left: 4px solid #10b981; }
                    .profit-value { font-size: 32px; font-weight: bold; color: #10b981; }
                    table { width: 100%%; border-collapse: collapse; margin: 20px 0; }
                    th, td { padding: 10px; text-align: left; border-bottom: 1px solid #e5e7eb; }
                </style>
            </head>
            <body>
                <div class="container">
                    <div class="header">
                        <h1>Daily Business Summary</h1>
                        <h2>%s</h2>
                        <p>%s</p>
                    </div>
                    
                    <div class="stats-grid">
                        <div class="stat-card">
                            <h3>%d</h3>
                            <p>Total Orders</p>
                        </div>
                        <div class="stat-card">
                            <h3>₹%.2f</h3>
                            <p>Total Revenue</p>
                        </div>
                        <div class="stat-card">
                            <h3>%d</h3>
                            <p>Completed Orders</p>
                        </div>
                        <div class="stat-card">
                            <h3>₹%.2f</h3>
                            <p>Average Order Value</p>
                        </div>
                    </div>
                    
                    <div class="profit-section">
                        <h3>Today's Profit</h3>
                        <div class="profit-value">₹%.2f</div>
                        <p>Profit Margin: %.1f%%</p>
                    </div>
                    
                    <h3>Top Selling Items</h3>
                    <table>
                        <thead>
                            <tr>
                                <th>Item</th>
                                <th>Quantity</th>
                                <th>Revenue</th>
                            </tr>
                        </thead>
                        <tbody>%s</tbody>
                    </table>
                    
                    <p style="text-align: center; color: #6b7280; margin-top: 40px;">
                        This is an automated daily summary from %s
                    </p>
                </div>
            </body>
            </html>
            """, data.getShopName(), data.getDate(), data.getTotalOrders(),
            data.getTotalRevenue(), data.getCompletedOrders(), 
            data.getAverageOrderValue(), data.getTotalProfit(),
            data.getProfitMargin(), topItemsHtml.toString(), appName);
    }

    // Helper methods
    private String getOTPSubject(String purpose) {
        return switch (purpose) {
            case "LOGIN" -> "Login OTP";
            case "SIGNUP" -> "Sign Up OTP";
            case "RESET_PASSWORD" -> "Password Reset OTP";
            case "ORDER_PICKUP" -> "Order Pickup OTP";
            case "ORDER_DELIVERY" -> "Order Delivery OTP";
            default -> "Verification OTP";
        };
    }

    private String getPurposeMessage(String purpose) {
        return switch (purpose) {
            case "LOGIN" -> "Use this OTP to login to your account";
            case "SIGNUP" -> "Use this OTP to complete your registration";
            case "RESET_PASSWORD" -> "Use this OTP to reset your password";
            case "ORDER_PICKUP" -> "Share this OTP with delivery partner for order pickup";
            case "ORDER_DELIVERY" -> "Share this OTP with delivery partner upon delivery";
            default -> "Use this OTP for verification";
        };
    }

    private String getStatusTitle(String status) {
        return switch (status) {
            case "CONFIRMED" -> "Order Confirmed";
            case "PREPARING" -> "Order Being Prepared";
            case "READY_FOR_PICKUP" -> "Ready for Pickup";
            case "OUT_FOR_DELIVERY" -> "Out for Delivery";
            case "DELIVERED" -> "Delivered Successfully";
            case "CANCELLED" -> "Order Cancelled";
            default -> "Order Update";
        };
    }

    private String generateStatusUpdateEmail(Map<String, Object> variables) {
        // Implementation for status update email
        return ""; // Simplified for brevity
    }

    private String generateDeliveryNotificationEmail(Map<String, Object> variables) {
        // Implementation for delivery notification email
        return ""; // Simplified for brevity
    }
}

// Data classes
@lombok.Data
@lombok.Builder
class InvoiceData {
    private String orderNumber;
    private String customerName;
    private String customerEmail;
    private String shopName;
    private List<InvoiceItem> items;
    private BigDecimal subtotal;
    private BigDecimal deliveryFee;
    private BigDecimal discount;
    private BigDecimal total;
    private String deliveryAddress;
    private LocalDateTime orderDate;
    private LocalDateTime deliveryDate;
}

@lombok.Data
class InvoiceItem {
    private String productName;
    private Integer quantity;
    private BigDecimal price;
    private String unit;
    private BigDecimal total;
}

@lombok.Data
@lombok.Builder
class DailySummaryData {
    private Long shopId;
    private String shopName;
    private String shopOwnerEmail;
    private String date;
    private Integer totalOrders;
    private Integer completedOrders;
    private Integer cancelledOrders;
    private Integer pendingOrders;
    private BigDecimal totalRevenue;
    private BigDecimal totalCost;
    private BigDecimal totalProfit;
    private BigDecimal profitMargin;
    private List<TopSellingItem> topSellingItems;
    private List<OrderDetail> orderDetails;
    private BigDecimal averageOrderValue;
    private String peakHours;
}

@lombok.Data
class TopSellingItem {
    private String name;
    private Integer quantity;
    private BigDecimal revenue;
}

@lombok.Data
class OrderDetail {
    private String orderNumber;
    private String customerName;
    private String items;
    private BigDecimal total;
    private String status;
    private String time;
}

File: frontend/src/app/core/services/email.service.ts
import { Injectable } from '@angular/core';
import { HttpClient } from '@angular/common/http';
import { Observable, of } from 'rxjs';
import { catchError, switchMap, tap } from 'rxjs/operators';
import { environment } from '../../../environments/environment';
import Swal from 'sweetalert2';

export interface EmailInvoiceRequest {
  orderNumber: string;
  customerEmail: string;
  customerName: string;
  shopName: string;
  items: InvoiceItem[];
  subtotal: number;
  deliveryFee: number;
  discount: number;
  total: number;
  deliveryAddress: string;
  orderDate: string;
  deliveryDate?: string;
}

export interface InvoiceItem {
  productName: string;
  quantity: number;
  price: number;
  unit: string;
  total: number;
}

export interface OTPRequest {
  email: string;
  purpose: 'LOGIN' | 'SIGNUP' | 'RESET_PASSWORD' | 'VERIFY_EMAIL' | 'ORDER_PICKUP' | 'ORDER_DELIVERY';
  orderId?: number;
}

export interface DailySummaryData {
  shopId: number;
  shopName: string;
  shopOwnerEmail: string;
  date: string;
  totalOrders: number;
  completedOrders: number;
  cancelledOrders: number;
  pendingOrders: number;
  totalRevenue: number;
  totalCost: number;
  totalProfit: number;
  profitMargin: number;
  topSellingItems: Array<{
    name: string;
    quantity: number;
    revenue: number;
  }>;
  orderDetails: Array<{
    orderNumber: string;
    customerName: string;
    items: string;
    total: number;
    status: string;
    time: string;
  }>;
  averageOrderValue: number;
  peakHours: string;
}

@Injectable({
  providedIn: 'root'
})
export class EmailService {
  private apiUrl = `${environment.apiUrl}/email`;

  constructor(private http: HttpClient) {}

  // Send OTP for authentication
  sendOTP(email: string, purpose: OTPRequest['purpose'], orderId?: number): Observable<{success: boolean, message: string}> {
    const request: OTPRequest = { email, purpose, orderId };
    
    return this.http.post<{success: boolean, message: string}>(`${this.apiUrl}/send-otp`, request).pipe(
      tap(response => {
        if (response.success) {
          let message = 'A 6-digit OTP has been sent to your email';
          if (purpose === 'ORDER_PICKUP') {
            message = 'Pickup OTP has been sent to shop owner';
          } else if (purpose === 'ORDER_DELIVERY') {
            message = 'Delivery OTP has been sent to customer';
          }
          Swal.fire({
            title: 'OTP Sent!',
            text: message,
            icon: 'success',
            timer: 3000
          });
        }
      }),
      catchError(error => {
        console.error('Error sending OTP:', error);
        return of({ success: false, message: 'Failed to send OTP' });
      })
    );
  }

  // Verify OTP
  verifyOTP(email: string, otp: string, purpose: string): Observable<{valid: boolean, token?: string}> {
    return this.http.post<{valid: boolean, token?: string}>(`${this.apiUrl}/verify-otp`, { email, otp, purpose }).pipe(
      catchError(error => {
        console.error('Error verifying OTP:', error);
        return of({ valid: false });
      })
    );
  }

  // Send invoice email with enhanced details
  sendInvoiceEmail(invoiceRequest: EmailInvoiceRequest): Observable<boolean> {
    const formData = new FormData();
    formData.append('orderNumber', invoiceRequest.orderNumber);
    formData.append('customerEmail', invoiceRequest.customerEmail);
    formData.append('invoiceHtml', this.generateInvoiceHtml(invoiceRequest));
    
    return this.http.post<{success: boolean}>(`${this.apiUrl}/send-invoice`, formData).pipe(
      switchMap(response => of(response.success)),
      tap(success => {
        if (success) {
          console.log(`Invoice email sent to ${invoiceRequest.customerEmail}`);
        }
      }),
      catchError(error => {
        console.error('Error sending invoice:', error);
        return of(false);
      })
    );
  }

  // Send order confirmation with OTP
  sendOrderConfirmationEmail(orderNumber: string, customerEmail: string, customerName: string, deliveryOTP?: string): Observable<boolean> {
    const emailData = {
      to: customerEmail,
      subject: `Order Confirmed - #${orderNumber}`,
      template: 'order-confirmation',
      data: {
        orderNumber,
        customerName,
        deliveryOTP,
        message: deliveryOTP ? `Your delivery OTP is: ${deliveryOTP}. Please share this with delivery partner upon delivery.` : ''
      }
    };
    
    return this.http.post<{success: boolean}>(`${this.apiUrl}/send`, emailData).pipe(
      switchMap(response => of(response.success)),
      catchError(() => of(false))
    );
  }

  // Send daily summary to shop owner
  sendDailySummary(summaryData: DailySummaryData): Observable<boolean> {
    const emailHtml = this.generateDailySummaryHtml(summaryData);
    
    const emailData = {
      to: summaryData.shopOwnerEmail,
      subject: `Daily Summary - ${summaryData.shopName} - ${summaryData.date}`,
      html: emailHtml,
      template: 'daily-summary'
    };
    
    return this.http.post<{success: boolean}>(`${this.apiUrl}/send`, emailData).pipe(
      tap(response => {
        if (response.success) {
          console.log(`Daily summary sent to ${summaryData.shopOwnerEmail}`);
        }
      }),
      switchMap(response => of(response.success)),
      catchError(error => {
        console.error('Error sending daily summary:', error);
        return of(false);
      })
    );
  }

  // Send notification when order is accepted/rejected
  sendOrderStatusEmail(customerEmail: string, orderNumber: string, status: 'ACCEPTED' | 'REJECTED', message: string, shopOTP?: string): Observable<boolean> {
    const emailData = {
      to: customerEmail,
      subject: `Order ${orderNumber} - ${status === 'ACCEPTED' ? 'Accepted' : 'Rejected'}`,
      template: 'order-status',
      data: {
        orderNumber,
        status,
        message,
        shopOTP: status === 'ACCEPTED' ? shopOTP : undefined,
        otpMessage: shopOTP ? `Shop Pickup OTP: ${shopOTP}` : ''
      }
    };
    
    return this.http.post<{success: boolean}>(`${this.apiUrl}/send`, emailData).pipe(
      switchMap(response => of(response.success)),
      catchError(() => of(false))
    );
  }

  // Send delivery notification with OTP
  sendDeliveryNotificationEmail(orderNumber: string, customerEmail: string, trackingUrl: string, deliveryOTP: string): Observable<boolean> {
    const emailData = {
      to: customerEmail,
      subject: `Your order ${orderNumber} is out for delivery!`,
      template: 'delivery-notification',
      data: {
        orderNumber,
        trackingUrl,
        deliveryOTP,
        message: `Your delivery OTP is: ${deliveryOTP}. Please share this with the delivery partner upon delivery.`
      }
    };
    
    return this.http.post<{success: boolean}>(`${this.apiUrl}/send`, emailData).pipe(
      switchMap(response => of(response.success)),
      catchError(() => of(false))
    );
  }

  generateInvoiceHtml(invoiceRequest: EmailInvoiceRequest): string {
    return `
      <!DOCTYPE html>
      <html>
      <head>
        <style>
          body { font-family: Arial, sans-serif; margin: 0; padding: 20px; background: #f5f5f5; }
          .container { max-width: 600px; margin: 0 auto; background: white; border-radius: 8px; overflow: hidden; box-shadow: 0 2px 10px rgba(0,0,0,0.1); }
          .header { background: #2563eb; color: white; padding: 20px; text-align: center; }
          .content { padding: 20px; }
          .order-info { background: #f8fafc; padding: 15px; border-radius: 6px; margin-bottom: 20px; }
          .items-table { width: 100%; border-collapse: collapse; margin-bottom: 20px; }
          .items-table th, .items-table td { padding: 10px; border-bottom: 1px solid #e5e7eb; text-align: left; }
          .items-table th { background: #f9fafb; font-weight: 600; }
          .total-section { background: #f0f9f0; padding: 15px; border-radius: 6px; }
          .total-row { display: flex; justify-content: space-between; margin-bottom: 5px; }
          .total-final { font-weight: bold; font-size: 18px; color: #059669; border-top: 2px solid #059669; padding-top: 10px; margin-top: 10px; }
          .footer { background: #f8fafc; padding: 20px; text-align: center; color: #6b7280; }
        </style>
      </head>
      <body>
        <div class="container">
          <div class="header">
            <h1>NammaOoru Delivery</h1>
            <h2>Order Invoice</h2>
          </div>
          <div class="content">
            <div class="order-info">
              <h3>Order Details</h3>
              <p><strong>Order Number:</strong> ${invoiceRequest.orderNumber}</p>
              <p><strong>Customer:</strong> ${invoiceRequest.customerName}</p>
              <p><strong>Shop:</strong> ${invoiceRequest.shopName}</p>
              <p><strong>Order Date:</strong> ${new Date(invoiceRequest.orderDate).toLocaleDateString('en-IN')}</p>
              ${invoiceRequest.deliveryDate ? `<p><strong>Delivery Date:</strong> ${new Date(invoiceRequest.deliveryDate).toLocaleDateString('en-IN')}</p>` : ''}
              <p><strong>Delivery Address:</strong> ${invoiceRequest.deliveryAddress}</p>
            </div>
            
            <h3>Items Ordered</h3>
            <table class="items-table">
              <thead>
                <tr>
                  <th>Item</th>
                  <th>Qty</th>
                  <th>Price</th>
                  <th>Total</th>
                </tr>
              </thead>
              <tbody>
                ${invoiceRequest.items.map(item => `
                  <tr>
                    <td>${item.productName}</td>
                    <td>${item.quantity} ${item.unit}</td>
                    <td>₹${item.price}</td>
                    <td>₹${item.total}</td>
                  </tr>
                `).join('')}
              </tbody>
            </table>
            
            <div class="total-section">
              <div class="total-row">
                <span>Subtotal:</span>
                <span>₹${invoiceRequest.subtotal}</span>
              </div>
              ${invoiceRequest.discount > 0 ? `
                <div class="total-row">
                  <span style="color: #059669;">Discount:</span>
                  <span style="color: #059669;">-₹${invoiceRequest.discount}</span>
                </div>
              ` : ''}
              <div class="total-row">
                <span>Delivery Fee:</span>
                <span>₹${invoiceRequest.deliveryFee}</span>
              </div>
              <div class="total-row total-final">
                <span>Total Amount:</span>
                <span>₹${invoiceRequest.total}</span>
              </div>
            </div>
            
            <p style="margin-top: 20px; color: #6b7280; font-style: italic;">
              Thank you for ordering with NammaOoru! We hope you enjoyed your meal.
            </p>
          </div>
          <div class="footer">
            <p>NammaOoru Multi-Service Delivery Platform</p>
            <p>Contact: +91 98765 43210 | Email: support@nammaooru.com</p>
          </div>
        </div>
      </body>
      </html>
    `;
  }

  // Generate daily summary HTML for shop owners
  generateDailySummaryHtml(data: DailySummaryData): string {
    return `
      <!DOCTYPE html>
      <html>
      <head>
        <style>
          body { font-family: Arial, sans-serif; margin: 0; padding: 20px; background: #f5f5f5; }
          .container { max-width: 800px; margin: 0 auto; background: white; border-radius: 8px; overflow: hidden; box-shadow: 0 2px 10px rgba(0,0,0,0.1); }
          .header { background: linear-gradient(135deg, #667eea 0%, #764ba2 100%); color: white; padding: 30px; text-align: center; }
          .content { padding: 30px; }
          .stats-grid { display: grid; grid-template-columns: repeat(2, 1fr); gap: 20px; margin-bottom: 30px; }
          .stat-card { background: #f8fafc; padding: 20px; border-radius: 8px; border-left: 4px solid #667eea; }
          .stat-value { font-size: 32px; font-weight: bold; color: #1a202c; }
          .stat-label { color: #718096; margin-top: 5px; }
          .profit-section { background: #f0fdf4; padding: 20px; border-radius: 8px; border-left: 4px solid #10b981; margin-bottom: 30px; }
          .profit-value { font-size: 36px; font-weight: bold; color: #10b981; }
          .table { width: 100%; border-collapse: collapse; margin-top: 20px; }
          .table th, .table td { padding: 12px; text-align: left; border-bottom: 1px solid #e5e7eb; }
          .table th { background: #f9fafb; font-weight: 600; color: #4b5563; }
          .top-items { background: #fef3c7; padding: 20px; border-radius: 8px; margin-bottom: 30px; }
          .footer { background: #f8fafc; padding: 20px; text-align: center; color: #6b7280; border-top: 1px solid #e5e7eb; }
        </style>
      </head>
      <body>
        <div class="container">
          <div class="header">
            <h1>Daily Business Summary</h1>
            <h2>${data.shopName}</h2>
            <p style="margin: 0; opacity: 0.9;">${data.date}</p>
          </div>
          
          <div class="content">
            <div class="stats-grid">
              <div class="stat-card">
                <div class="stat-value">${data.totalOrders}</div>
                <div class="stat-label">Total Orders</div>
              </div>
              <div class="stat-card">
                <div class="stat-value">₹${data.totalRevenue.toLocaleString()}</div>
                <div class="stat-label">Total Revenue</div>
              </div>
              <div class="stat-card">
                <div class="stat-value">${data.completedOrders}</div>
                <div class="stat-label">Completed Orders</div>
              </div>
              <div class="stat-card">
                <div class="stat-value">₹${data.averageOrderValue.toFixed(0)}</div>
                <div class="stat-label">Average Order Value</div>
              </div>
            </div>
            
            <div class="profit-section">
              <h3 style="margin-top: 0;">Today's Profit</h3>
              <div class="profit-value">₹${data.totalProfit.toLocaleString()}</div>
              <p style="color: #6b7280; margin: 10px 0;">Profit Margin: ${data.profitMargin.toFixed(1)}%</p>
              <p style="color: #6b7280; margin: 0;">Total Cost: ₹${data.totalCost.toLocaleString()}</p>
            </div>
            
            <div class="top-items">
              <h3 style="margin-top: 0;">Top Selling Items</h3>
              <table class="table">
                <thead>
                  <tr>
                    <th>Item</th>
                    <th>Quantity Sold</th>
                    <th>Revenue</th>
                  </tr>
                </thead>
                <tbody>
                  ${data.topSellingItems.slice(0, 5).map(item => `
                    <tr>
                      <td>${item.name}</td>
                      <td>${item.quantity}</td>
                      <td>₹${item.revenue}</td>
                    </tr>
                  `).join('')}
                </tbody>
              </table>
            </div>
            
            <div>
              <h3>Order Details</h3>
              <table class="table">
                <thead>
                  <tr>
                    <th>Order #</th>
                    <th>Customer</th>
                    <th>Items</th>
                    <th>Total</th>
                    <th>Status</th>
                    <th>Time</th>
                  </tr>
                </thead>
                <tbody>
                  ${data.orderDetails.map(order => `
                    <tr>
                      <td>${order.orderNumber}</td>
                      <td>${order.customerName}</td>
                      <td>${order.items}</td>
                      <td>₹${order.total}</td>
                      <td>${order.status}</td>
                      <td>${order.time}</td>
                    </tr>
                  `).join('')}
                </tbody>
              </table>
            </div>
            
            <div style="margin-top: 30px; padding: 20px; background: #eff6ff; border-radius: 8px;">
              <h4 style="margin-top: 0;">Business Insights</h4>
              <p>• Peak hours: ${data.peakHours}</p>
              <p>• Pending orders: ${data.pendingOrders}</p>
              <p>• Cancelled orders: ${data.cancelledOrders}</p>
            </div>
          </div>
          
          <div class="footer">
            <p style="margin: 0;">This is an automated daily summary from NammaOoru</p>
            <p style="margin: 10px 0 0 0;">For support: support@nammaooru.com | +91 98765 43210</p>
          </div>
        </div>
      </body>
      </html>
    `;
  }
}